packages/gatsby-theme-store/src/sdk/search/controller.ts

```diff
@@ -53,7 +53,25 @@ export const setSearchFilters = (filters: SearchFilters) => {
     }
   }
 
-  navigate(`/${filters.query}?${params.toString()}`)
+  /**
+   * Usually, "filters.query" is the right place to navigate to. However, in landing pages
+   * the "filters.query" contains a different value from the page's root path because the
+   * business user decides the value of "canonicalPath". This makes us have to always
+   * add the root path to the navigation so we never miss the landing page navigation context
+   */
+  const spath = pathname.split('/').slice(1)
+  const squery = filters.query.split('/')
+
+  let it = 0
+
+  while (it < spath.length && spath[it] !== squery[0]) {
+    it++
+  }
```

Comment:
I think changing to this will make it easier to read:

```suggestion
  const it = spath.findIndex((path) => squery[0] === path) ?? 0
```